hs_core/templates/search/indexes/hs_core/baseresource_text.txt

```diff
@@ -1,60 +1,111 @@
-{{ object.short_id }}
-{{ object.doi }}
-{{ object.metadata.title.value }}
-{{ object.first_creator }}
-{{ object.metadata.description }}
+{% if object.short_id %} 
+    {{ object.short_id }}
+{% endif %} 
+{% if object.doi %} 
+    {{ object.doi }}
+{% endif %} 
+{% if object.metadata.title.value %}
+    {{ object.metadata.title.value }}
+{% endif %} 
+{% if object.metadata.description %} 
+    {{ object.metadata.description }}
+{% endif %} 
 {{ object.public }}
 {{ object.discoverable }}
 {{ object.published }}
 {{ object.created }}
 {{ object.modified }}
-{{ object.rating_sum }}
-{{ object.metadata.publisher.name }}
-{{ object.metadata.language.code }}
-{{ object.resource_type }}
-{{ object.verbose_name }}
-{{ object.owners_count }}
-{{ object.viewers_count }}
-{{ object.editors_count }}
-{{ object.comments_count }}
+{% if object.rating_sum %} 
+    {{ object.rating_sum }}
+{% endif %} 
+{% if object.metadata.publisher.name %} 
```

Comment:
See https://docs.djangoproject.com/en/1.9/ref/templates/api/#how-invalid-variables-are-handled
In general, any undefined template variable will be treated as None (False). 
The statement {% if a.b.c.d %} 
means 
{% if a and a.b and a.b.c and a.b.c.d %}